### ISSUE TITLE
Confusing examples of package version spec string.

### ISSUE THREAD

**Issue by rainwoodman** [conda/conda-docs#448]
The documentation claims 'python>=2.7' matches a package named 'python>=2.7' But then the following examples with 'numpy>=1.8' suggests it actually works.

https://github.com/conda/conda-docs/blob/master/docs/source/spec.rst#package-match-specifications

Is a space after package name needed or not?

**Comment by caleb-easterly** [comment]
Yes, I've been confused about this too. The documentation does not clearly specify what to do when a package version needs to match exactly, either.

**Comment by jakirkham** [comment]
AFAICT cb3 made this optional. Is that correct @msarahan?

**Comment by msarahan** [comment]
I believe spaces in package specs are mandatory.  The version spec stuff is horribly fractured across the many places that you might want to specify versions.  The general syntax holds, but you do need to put spaces in between the package name, version constraint (comma separated for AND, | separated for OR), and build string.

One day we'll try to unify these everywhere, but there are bigger fish to fry for the time being.

**Comment by jakirkham** [comment]
So you might find PR ( https://github.com/conda-forge/holoviews-feedstock/pull/33 ) interesting then. Seems dropping spaces was ok there on cb3.

**Comment by asmeurer** [comment]
Note there is a difference between the package spec (what goes in the .tar.bz2) and the conda-build meta.yaml. conda-build may be more lenient in what it allows, translating things for the final spec (I don't know if that is the case here).

**Comment by msarahan** [comment]
Yes, @asmeurer is right, there is re-interpretation in many cases.  Dropping spaces might work, but I think it will also be fragile.  We make a ton of assumption based on splitting a spec.  If a split with no args returns 1 thing, it's just a package name.  With 2 things, the second is the version constraint.  With 3, the third is the build string.  Making that no longer true will almost certainly break stuff.

**Comment by msarahan** [comment]
PS: there is a space in the spec: https://github.com/conda-forge/holoviews-feedstock/pull/33/files#diff-e178b687b10a71a3348107ae3154e44cR28

The comma-separated stuff in the version spec should not have a space.  The space between the name and the version spec is required.

**Comment by jakirkham** [comment]
Right, that change was made after discussion here. Commit ( https://github.com/conda-forge/holoviews-feedstock/pull/33/commits/6a388d2a5c5fc59f130293d3c2d6d4fb15d1413d ) is relevant to the original point, which fails on cb2, but passed after re-rendering (IOW using cb3).

**Comment by jakirkham** [comment]
Have raised issue ( https://github.com/conda/conda-build/issues/2828 ) about the lack of space being accepted in cb3.

**Comment by asmeurer** [comment]
Allowing no spaces in conda-build is probably fine. There aren't really packages with inequality symbols in their names, at least as far as I know. I would suggest keeping the package spec itself as simple as possible, though.

**Comment by rrigdon** [comment]
From the docs: 

> Remember that the version specification cannot contain spaces, as spaces are used to delimit the package, version, and build string in the whole match specification. python >= 2.7 is an invalid match specification. Furthermore, python>=2.7 is matched as any version of a package named python>=2.7.

https://docs.conda.io/projects/conda-build/en/latest/resources/package-spec.html

@msarahan - is the above statement true or are spaces required in pkg specifications?
